### ISSUE TITLE
`GeoPath.intersect` should return `GeoPoint`

### ISSUE THREAD

**Issue by kastork** [pbrod/nvector#16]
The intersect method on `GeoPath` objects claims to return an instance of `GeoPoint`, which seems like the desired behavior (at least, that is what I want it to return too).  However it returns an instance of `Nvector`.

From the last line of the `intersect` method...

`return frame.Nvector(normal_c, z=depth)`

**Comment by mhogan-nwra** [comment]
Hello @kastork 

I have taken the liberty to fork the nvector package into a new PyPI package titled [envector](https://pypi.org/project/envector/). The envector package is intended as a drop-in replacement with minimal changes in your source code as no methods and classes have been removed since the fork. For example, this is a valid snippet

import envector as nv
wgs84 = nv.FrameE(name='WGS84')

As for your issue, I also caught that mistake when I updated the docstrings and add type-hints. If you examine the 
[envector.objects.GeoPath.intersect](https://github.com/mhogan-nwra/envector/blob/8a537f6ad16ccc6205ed9b9937f4f50132c9aa0b/src/envector/objects.py#L1427) method, you will find the correction you suggested. I also re-rendered its public [documentation pages](https://envector.readthedocs.io/en/latest/reference/nvector_summary.html#envector.objects.GeoPath.intersect). I am little displeased with the type-hint being a private super-class, but the documentation is better at least.
